sample: Bluetooth: HFP_AG: Support voice transmission and playback

lylezhu2012 opened the pull request:
Add PCM and CODEC support for voice streaming in the HFP AG sample.

Implement PCM interface to handle voice data transmission between the Bluetooth controller and the application using I2S driver.

Implement CODEC interface to handle voice playback and capture using I2S and audio codec drivers.

Add bidirectional voice streaming: capture voice from CODEC and transmit through PCM when SCO connection is established, and receive voice from PCM and playback through CODEC.

Initialize PCM and CODEC interfaces based on the SCO air mode (CVSD or mSBC) when SCO connection is established.

Add codec negotiation callbacks to support codec selection.

Enable codec negotiation and mSBC codec support for mimxrt1170_evk board.

Add configuration options for audio transfer interval, buffer counts, and thread priorities for both PCM and CODEC RX threads.

Update documentation with voice streaming topology diagram and running instructions.

lylezhu2012 committed:
sample: Bluetooth: HFP_AG: Support voice transmission and playback

Add PCM and CODEC support for voice streaming in the HFP AG sample.

Implement PCM interface to handle voice data transmission between the
Bluetooth controller and the application using I2S driver.

Implement CODEC interface to handle voice playback and capture using
I2S and audio codec drivers.

Add bidirectional voice streaming: capture voice from CODEC and
transmit through PCM when SCO connection is established, and receive
voice from PCM and playback through CODEC.

Initialize PCM and CODEC interfaces based on the SCO air mode (CVSD or
mSBC) when SCO connection is established.

Add codec negotiation callbacks to support codec selection.

Enable codec negotiation and mSBC codec support for mimxrt1170_evk
board.

Add configuration options for audio transfer interval, buffer counts,
and thread priorities for both PCM and CODEC RX threads.

Update documentation with voice streaming topology diagram and running
instructions.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

MarkWangChinese commented on samples/bluetooth/classic/handsfree_ag/README.rst:
```diff
@@ -19,3 +19,79 @@ Building and Running
+           AG [label="HFP HF"];
+       }
+
+       subgraph cluster_hf {
```
```Suggestion
       subgraph cluster_ag {
```

lylezhu2012 replied:
Fixed.

makeshi commented on samples/bluetooth/classic/handsfree_ag/src/main.c:
```diff
@@ -23,8 +23,12 @@
+#include "codec.h"
+
 static struct bt_conn *default_conn;
 struct bt_hfp_ag *hfp_ag;
```
Need static?

lylezhu2012 committed:
sample: Bluetooth: HFP_AG: Make global variables `static`

Make global variables static in the HFP AG sample to improve
encapsulation and avoid potential symbol conflicts.

Change the following variables from global to static scope:
- hfp_ag
- hfp_ag_call
- discover_work
- call_connect_work
- call_disconnect_work
- call_remote_ringing_work
- call_remote_accept_work

These variables are only used within main.c and do not need to be
externally visible.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>
